sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -185,8 +183,21 @@ case class DataSource(
     }
 
     SchemaUtils.checkColumnNameDuplication(
```

Comment:
sorry I made a mistake, shall we move these 2 checks in constructor? I think they are useful even for non-file based data sources